packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -566,7 +566,18 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
       } catch (err) {
         const isAlreadyKnown = /AlreadyKnown/gi.test(err.message)
         const isFeeTooLow = /FeeTooLowToCompete/gi.test(err.message)
+        const nonceTooLow = /(nonce.*too low|already been used|NONCE_EXPIRED)/gi.test(err.message)
         const shouldRetry = (isAlreadyKnown || isFeeTooLow) && i < maxRetries
+        if (nonceTooLow) {
+          if (!this.nonce) {
+            // wait a bit before attempting again so it re-fetches latest nonce
+            const warnMsg = `nonce too low. Waiting a moment before retrying again. Error: ${err.message}`
+            this.logger.warn(warnMsg)
+            this.notifier.warn(warnMsg, { channel: gasBoostWarnSlackChannel })
+            await wait(10 * 1000)
+            continue
```

Comment:
I think we discussed only trying 5 times and then erroring or throwing away the transfer. If I am reading this correctly, I think it would constantly try a single tx every 10s forever until it works. This would block all other transactions, I believe.

I think we need to throw it away or do something different so we don't get stuck in an infinite loop. I'd expect an RPC endpoint to fix itself eventually, but if we have a bug in our code that would cause this to fail every time, I'd imagine we need to not loop infinitely.

Reply:
I believe it is fixed with https://github.com/hop-protocol/hop/pull/178/commits/c3439f45339a4c385de5cfffbaf617c8f9873441